### ISSUE TITLE
Unable to build SciPy with lapack-reference

### ISSUE THREAD

**Issue by RaitoBezarius** [NixOS/nixpkgs#86613]
**Describe the bug**
It's impossible to build SciPy (or NumPy) with netlib.org's BLAS currently using the new BLAS/LAPACK switching mechanism.

SciPy won't build and NumPy does not detect any BLAS/LAPACK.

**To Reproduce**
1. Pin yourself to `nixos-unstable-small` (rev: `3b00e78e63b6539369821660245cd579f6723b58"`, latest available at the moment.)
2. Switch using LAPACK/BLAS switching mechanism to `lapack-reference` (i.e. netlib.org)
3. Build: `python.withPackages (ps: [ps.numpy ps.scipy])`.

Quick `nix-shell` snippet:
```nix
let ourNixpkgs = 
  builtins.fetchGit {
  # Descriptive name to make the store path easier to identify
  name = "nixos-unstable-small-2020-05-02";
  url = "https://github.com/nixos/nixpkgs-channels/";
  # Commit hash for nixos-unstable as of 2020-05-02
  # `git ls-remote https://github.com/nixos/nixpkgs-channels nixos-unstable`
  ref = "refs/heads/nixos-unstable-small";
  rev = "3b00e78e63b6539369821660245cd579f6723b58";
};
in
{ blasProvider ? "openblasCompat"
, lapackProvider ? blasProvider
, pkgs ? import ourNixpkgs {
  config.allowUnfree = blasProvider == "mkl";
  overlays = [(self: super: {
      blas = super.blas.override {
        blasProvider = super.${blasProvider};
      };
      lapack = super.lapack.override {
        lapackProvider = super.${lapackProvider};
      };
    })];
}
, pythonPackageName ? "python3"
, python ? pkgs.${pythonPackageName}}:
rec {
  pythonDependencies = (python.withPackages
  (ps: [
    ps.numpy
    ps.scipy
  ]));

  shell = pkgs.mkShell {
    buildInputs = with pkgs; [
      pythonDependencies
    ];
  };
};
```

Now, you can test with `nix-shell $THE_FILE -A shell --argstr blasProvider "lapack-reference"`.

**Expected behavior**
An environment with SciPy/NumPy compiled against netlib.org's BLAS.

**Additional context**
Recent PRs have been made (#85799) notably to fix ongoing issues with BLAS, that might be a regression or this scenario has not been tested yet.

**Notify maintainers**
@matthewbauer @FRidh @bhipple (according to what I've seen on the PRs).

**Metadata**
(not really relevant but well, I follow instructions)
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.6.8-arch1-1, Arch Linux, noversion`
 - multi-user?: `yes`
 - sandbox: `relaxed`
 - version: `nix-env (Nix) 2.3.4`
 - channels(root): `"nixpkgs-20.09pre221715.5f814bf9cc4"`
 - channels(raito): `"nixpkgs-20.09pre221715.5f814bf9cc4"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

Maintainer information:
```yaml
attribute: lapack-reference
```

**Comment by bhipple** [comment]
The two we've been looking at so far are `openblas` and `mkl`; I don't think the netlib scenario has been tested until now.

**Comment by RaitoBezarius** [comment]
@bhipple Is this something that can be supported?
I didn't succeed to recompile against openblas, but only openblasCompat due to the 32 bits vs 64 bits stuff I guess.

**Comment by bhipple** [comment]
`openblas` is the default on 64-bit, openblasCompat on 32:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/alternatives/lapack/default.nix#L4

This is what Hydra is currently testing for scipy.  From my end I'm just using either openblas or mkl, but if you're willing to support the netlib blas I'd be happy to review/test a PR.  What's the draw over just using openblas?

**Comment by RaitoBezarius** [comment]
> `openblas` is the default on 64-bit, openblasCompat on 32:
> https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/alternatives/lapack/default.nix#L4
> 
> This is what Hydra is currently testing for scipy. From my end I'm just using either openblas or mkl, but if you're willing to support the netlib blas I'd be happy to review/test a PR. What's the draw over just using openblas?

I'm doing cross-tests between those implementations and their impact on the end-results.
I know very well that netlib is super slow, but my computations are not that heavy, I'm more interested into the correctness.

Strange @ openblas, I have to check again, I have a 64 bits system but openblasCompat is the only one working, when openblas is failing due to assert isLP64 something something.

**Comment by matthewbauer** [comment]
Sorry about this! I thought I was testing the lapack-reference case, but for some reason not in scipy/numpy. The issue is that the section we set in site.cfg is not the same as what numpy expects. Adding 

This should be fixed in https://github.com/NixOS/nixpkgs/pull/87220.

**Comment by RaitoBezarius** [comment]
> Sorry about this! I thought I was testing the lapack-reference case, but for some reason not in scipy/numpy. The issue is that the section we set in site.cfg is not the same as what numpy expects. Adding a lapack/blas section covers this case.
> 
> This should be fixed in #87220.

Awesome!

**Comment by FRidh** [comment]
Fix is in staging-next. Should land in master in a couple of days.
